#!/bin/sh

#  ShortStack_allNewSmallRNADE.sh
#
#
#  Created by THC Ribeiro on 21/02/2022.
#
#get the right files

cp /home/tcherubino/Carabica.smallRNA/DE

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5540_chopped.txt	G5_Cat_green_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5541_chopped.txt	G5_Cat_green_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5542_chopped.txt	G5_Cat_green_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5543_chopped.txt	G6_Cat_white_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5544_chopped.txt	G6_Cat_white_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5545_chopped.txt	G6_Cat_white_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5546_chopped.txt	Flower_Cat_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5553_chopped.txt	G1_Cat_IVG_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5554_chopped.txt	G1_Cat_IVG_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5555_chopped.txt	G1_Cat_IVG_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5556_chopped.txt	G2_Cat_IN_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5557_chopped.txt	G2_Cat_IN_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5558_chopped.txt	G3_Cat_bud3_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5559_chopped.txt	G3_Cat_bud3_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5560_chopped.txt	G3_Cat_bud3_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5561_chopped.txt	G4_Cat_bud36e_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5562_chopped.txt	G4_Cat_bud36e_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5563_chopped.txt	G4_Cat_bud36e_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5564_chopped.txt	G4_Cat_bud36l_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5565_chopped.txt	G4_Cat_bud36l_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5566_chopped.txt	G4_Cat_bud36l_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5567_chopped.txt	Flower_Cat_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5568_chopped.txt	Flower_Cat_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5569_chopped.txt	G1_Sir_IVG_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5570_chopped.txt	G1_Sir_IVG_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5571_chopped.txt	G1_Sir_IVG_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5572_chopped.txt	G2_Sir_IN_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5573_chopped.txt	G2_Sir_IN_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5574_chopped.txt	G2_Sir_IN_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5575_chopped.txt	G3_Sir_bud3_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5576_chopped.txt	G3_Sir_bud3_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5577_chopped.txt	G3_Sir_bud3_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5578_chopped.txt	G4_Sir_bud36e_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5579_chopped.txt	G4_Sir_bud36e_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5580_chopped.txt	G4_Sir_bud36e_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5581_chopped.txt	G4_Sir_bud36l_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5582_chopped.txt	G4_Sir_bud36l_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5583_chopped.txt	G4_Sir_bud36l_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5584_chopped.txt	G5_Sir_green_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5585_chopped.txt	G5_Sir_green_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5586_chopped.txt	G5_Sir_green_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5587_chopped.txt	G6_Sir_white_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5588_chopped.txt	G6_Sir_white_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5589_chopped.txt	G6_Sir_white_r3.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5590_chopped.txt	Flower_Sir_r1.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5591_chopped.txt	Flower_Sir_r2.fasta &

python /alldata/tagcount_to_fasta_decondense.py ../smallRNAsLibraries/5592_chopped.txt	Flower_Sir_r3.fasta &


##########################
#########create unified###
#####Small Reference######
##########################

#list all gff elements

awk '{print $3}' GCF_003713225.1_Cara_1.0_genomic.gff | sort -u

#rRNA (not to be used, but good to keep
awk '{ if ($3=="rRNA") print $0 }' GCF_003713225.1_Cara_1.0_genomic.gff > rRNA.gff


bedtools getfasta -fi all.genome.fasta -fo Carabica.rRNA.fasta -bed rRNA.gff

#include snRNA in header
sed 's/>/>rRNA_/g' Carabica.rRNA.fasta > temp

#all lower case to upper case
sed -e 's/\(.*\)/\U\1/' temp >  Carabica.rRNA.fasta

#verify the presence of identical sequences
~/bin/genometools-1.6.1/bin/gt sequniq -o v Carabica.rRNA.fasta
# 911 out of 1643 sequences have been removed (55.447%)

mv v Carabica.rRNA.fasta

#snoRNA
awk '{ if ($3=="snoRNA") print $0 }' GCF_003713225.1_Cara_1.0_genomic.gff > snoRNAs.gff

bedtools getfasta -fi all.genome.fasta -fo Carabica.snoRNAs.fasta -bed snoRNAs.gff

#include snRNA in header
sed 's/>/>snoRNA_/g' Carabica.snoRNAs.fasta > temp

#all lower case to upper case
sed -e 's/\(.*\)/\U\1/' temp >  Carabica.snoRNAs.fasta

#verify the presence of identical sequences
~/bin/genometools-1.6.1/bin/gt sequniq -o v Carabica.snoRNAs.fasta
#265 out of 1468 sequences have been removed (18.052%)

mv v Carabica.snoRNAs.fasta

#snRNA

awk '{ if ($3=="snRNA") print $0 }' GCF_003713225.1_Cara_1.0_genomic.gff > snRNA.gff

bedtools getfasta -fi all.genome.fasta -fo Carabica.snRNA.fasta -bed snRNA.gff

#include snoRNA in header
sed 's/>/>snRNA_/g' Carabica.snRNA.fasta > temp

#all lower case to upper case
sed -e 's/\(.*\)/\U\1/' temp >  Carabica.snRNA.fasta

~/bin/genometools-1.6.1/bin/gt sequniq -force -o v Carabica.snRNA.fasta

# 20 out of 178 sequences have been removed (11.236%)
mv v Carabica.snRNA.fasta


#tRNA

awk '{ if ($3=="tRNA") print $0 }' GCF_003713225.1_Cara_1.0_genomic.gff > tRNA.gff

bedtools getfasta -fi all.genome.fasta -fo Carabica.tRNA.fasta -bed tRNA.gff


#include snoRNA in header
sed 's/>/>tRNA_/g' Carabica.tRNA.fasta > temp

#all lower case to upper case
sed -e 's/\(.*\)/\U\1/' temp >  Carabica.tRNA.fasta

~/bin/genometools-1.6.1/bin/gt sequniq -o v Carabica.tRNA.fasta

# 412 out of 719 sequences have been removed (57.302%)

mv v Carabica.tRNA.fasta

####
#remove duplication in PHAS RNAs
~/bin/genometools-1.6.1/bin/gt sequniq -o v PHAS.fa

#good, no duplication

#Colapse identical mature miRNA

mkdir smallReference
cd smallReference

mkdir process

python matureMirCollapser.py MatheusAndMiRadorMature.fasta identicalClusters.json > collapesed.C.a.Mature.Mir.fasta

cp collapesed.C.a.Mature.Mir.fasta ../collapesed.C.a.Mature.Mir.fasta

cd ..

cat Carabica.snRNA.fasta Carabica.snoRNAs.fasta collapesed.C.a.Mature.Mir.fasta PHAS.fa Carabica.tRNA.fasta > allSmall.fasta

mkdir ShortStack_alignment
cp ../smallReference/allSmall.fasta ./

samtools faidx allSmall.fasta
awk 'BEGIN {FS="\t"}; {print $1":1-"$2}' allSmall.fasta.fai > locus


cd ../ShortStack_alignment


mkdir w7

cd w7


cp ../allSmall.fasta ./

#grep ">" allSmall.fasta | sed 's/>//g' > locusFile

#awk 'BEGIN {FS="\t"}; {print $1":1-"$2}' allSmall.fasta.fai > locus

for file in `ls ../../*fasta`;
do

#all miRNA
#shortStack v 3.8.5
~/bin/ShortStack/ShortStack --g ../allSmall.fasta --bowtie_cores 60 --readfile $file --locifile ../locus --mmap u --nohp --mismatches 0 --outdir `basename -s fasta $file`twoMismatch.ShortStack
done

#calculate sequence length from fasta file
#will use then latter to produce RPKM values
awk '/^>/{if (l!="") print l; print; l=0; next}{l+=length($0)}END{print l}' ../allSmall.fasta | tr "\n" "\t" | tr ">" "\n" > sequencesLengths.tab

#sed '/miR/d' sequencesLengths.tab | sed '/Novel/d' > n

#mv n sequencesLengths.tab

mkdir ../shortStack

mv *twoMismatch.ShortStack ../shortStack/

# The results are in the 'Counts.txt' file, which has the observed number
#of reads, the mean number of reads for the ten re-samplings, and the
#standard deviations. When there are multiple read-groups, this is
#helpful to gather the raw data for differential expression analysis.

#There is always a read-group called 'main', which is all read-groups
#combined.

#for folder in `ls -d */`; do echo $folder; cd $folder; rm *bai; cd ..; done

#for folder in `ls -d */`; do echo $folder; cd $folder; rm *bam; cd ..; done

mkdir counts

cd counts

#the names say "twoMismatch" but, in fact, the files are
#no mismatches
for folder in `ls -d ../shortStack/*twoMismatch.ShortStack/`
do
    echo $folder
    cd $folder
    pwd
    cp Counts.txt ../../counts/`basename -s .twoMismatch.ShortStack $folder`.counts.tab
    cd ..
done

cd ../counts

#remove lines containing "NA"

#sed -i '/NA/d'  G6_Sir_white_r3.noMismatch.counts.tab

for file in `ls *tab`
do
  #Remove the unplaced smallRNAs
    awk '{if($2 != "NA") {print $1"\t"$3}}' $file > `basename -s fastq.counts.tab $file`counts.txt
    sed -i"" -e "1d" `basename -s fastq.counts.tab $file`counts.txt
done

rm *tab

#all files has NOT the same order
#now they have....
mkdir process
cd process

for file in $(ls ../*tabcounts.txt)
do
echo $file

python processShortStackcount.py $file > $(basename -s counts.tabcounts.txt $file)counts.tab


done


#create a directory to output the full report from shortStack.
#This data is imporant for some general analysis that will be carried out at "misc_for_w7_smallRNApipeline"


mkdir results

cd results

#the names say "twoMismatch" but, in fact, the files are
#no mismatches
for folder in `ls -d ../shortStack/*twoMismatch.ShortStack/`
do
    echo $folder
    cd $folder
    pwd
    cp Results.txt `basename -s .twoMismatch.ShortStack $folder`


    echo "lib" > library.id
    for i in {1..2948};
    do
      echo $(basename -s .twoMismatch.ShortStack $folder) >> library.id
    done

    paste library.id `basename -s .twoMismatch.ShortStack $folder` > ../../results/`basename -s .twoMismatch.ShortStack $folder`.results.tab


    cd ..
done

cd ../results

cat *tab > shortSatckAllResults.txt

#remove the header
sed -i"" '/Length/d' shortSatckAllResults.txt

#create a file containing the header

#lib     Locus  Name    Length  Reads   RPM     UniqueReads     FracTop Strand  MajorRNA        MajorRNAReads   Complexity      DicerCall       MIRNA   PhaseScore      Short   Long    20      21      22      23      24

cat header shortSatckAllResults.txt > t

mv t shortSatckAllResults.txt
##############################
# DE analysis################
##############################
R

library("edgeR")
targets <- readTargets()
names <- targets$description
matrix_input <- readDGE(targets, comment.char = "!")


reads_before <- sum(matrix_input$counts)
#remove low expressed genes
rnaseqmatrix <- matrix_input$counts
rnaseqmatrix <- rnaseqmatrix[rowMeans(rnaseqmatrix) >=5,]

#only in w3
#rnaseqmatrix <- rnaseqmatrix[rowMeans(rnaseqmatrix) >=5,]


conditions = matrix_input$samples[,2]

rownames(rnaseqmatrix) <- gsub("NC_039898.1","1",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039899.1","2",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039900.1","3",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039901.1","4",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039902.1","5",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039903.1","6",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039904.1","7",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039905.1","8",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039906.1","9",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039907.1","10",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039908.1","11",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039909.1","12",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039910.1","13",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039911.1","14",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039912.1","15",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039913.1","16",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039914.1","17",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039915.1","18",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039916.1","19",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039917.1","20",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039918.1","21",rownames(rnaseqmatrix))

rownames(rnaseqmatrix) <- gsub("NC_039919.1","22",rownames(rnaseqmatrix))

L24P.like <- read.delim("L24P.like.tab",header=F)
PHAS21 <- read.delim("21PHAS.tab",header=F)
PHAS24 <- read.delim("24PHAS.tab",header=F)
unknown <- read.delim("unknown.tab",header=F)

 rownames(rnaseqmatrix)[match(PHAS21$V1,rownames(rnaseqmatrix))] <- gsub("PHAS","21PHAS",rownames(rnaseqmatrix)[match(PHAS21$V1,rownames(rnaseqmatrix))])

 rownames(rnaseqmatrix)[match(PHAS24$V1,rownames(rnaseqmatrix))] <- gsub("PHAS","24PHAS",rownames(rnaseqmatrix)[match(PHAS24$V1,rownames(rnaseqmatrix))])

 rownames(rnaseqmatrix)[match(L24P.like$V1,rownames(rnaseqmatrix))] <- gsub("PHAS","L24P.like",rownames(rnaseqmatrix)[match(L24P.like$V1,rownames(rnaseqmatrix))])


  rownames(rnaseqmatrix)[match(unknown$V1,rownames(rnaseqmatrix))] <- gsub("PHAS","UNK",rownames(rnaseqmatrix)[match(unknown$V1,rownames(rnaseqmatrix))])


analysis_matrix <- DGEList(counts = rnaseqmatrix,group = conditions)
colnames(analysis_matrix$counts) <- names
design <- model.matrix(~0+group, data=analysis_matrix$samples)

colnames(design) <- levels(analysis_matrix$samples$group)

#NORMALIZATIONS
analysis_matrix <- calcNormFactors(analysis_matrix)

#To estimate common dispersion:
analysis_matrix <- estimateGLMCommonDisp(analysis_matrix, design)
#To estimate trended dispersions:
analysis_matrix <- estimateGLMTrendedDisp(analysis_matrix, design)
#To estimate tagwise dispersions:
analysis_matrix <- estimateGLMTagwiseDisp(analysis_matrix, design)

#Fit a negative binomial generalized log-linear model to the read counts for each gene.
fit <- glmFit(analysis_matrix,design)

pdf(file = "edgeR_BCV.pdf",h=8,w=8)
plotBCV(analysis_matrix)
dev.off()

system("open edgeR_BCV.pdf")

cols1 <- colorRampPalette(c("#5F6D54","#CAB393","#BF7F58"))(47)

#cols1 <- c(rep("#112512",24),rep("#486047",24),rep("#7F9B7D",24),rep("#B6D7B3",24))

#An MDS plots shows distances, in terms of biological coeficient of variation (BCV) - An MDS plot shows the relative similarities of the samples.
pdf(file = "edgeR_MDS.pdf",h=8,w=8)
plotMDS(analysis_matrix,col = cols1,cex=1.5)
dev.off()
system("open edgeR_MDS.pdf")

#samples_trees
cpm.matrix <- cpm(analysis_matrix,normalized.lib.sizes=F)
colnames(cpm.matrix) <- names
t.cpm.matrix <- t(cpm.matrix)
sampleTree <- hclust(dist(t.cpm.matrix), method = "average");

# Plot the sample tree: Open a graphic output window of size 12 by 9 inches
# The user should change the dimensions if the window is too large or too small.
#sizeGrWindow(12,9)
pdf(file = "sampleClustering.pdf", width = 12, height = 9);
par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
# Plot a line to show the cut
#abline(h = 15, col = "red")
dev.off()

system("open sampleClustering.pdf")

#analyse densisties before normalization - input
pdf("Densities_input.pdf")
plotDensities( log(matrix_input$counts), legend = "topright")
dev.off()

#analyse densisties low expression filter
pdf("Densities_low_expression_fiter.pdf")
plotDensities( log(rnaseqmatrix), legend = "topright")
dev.off()

#analyse densisties after normalization
pdf("Densities_normalization.pdf")
plotDensities( log(cpm.matrix), legend = "topright")
dev.off()

pdf("Densities_log_cpm_fitted_norm.pdf")
plotDensities(log(cpm(fit$fitted.values, normalized.lib.sizes=TRUE)), legend = "topright")
dev.off()

#histogram of densities log10
pdf("Log10_histogram_normilized.pdf", h=10,w=10)
hist(log(cpm.matrix+1,10), col=gray.colors(19, start = 0.9, end = 0.3))
dev.off()

#histogram of densities Log2
pdf("Log2_histogram_normilized.pdf", h=10,w=10)
hist(log(cpm.matrix+1,2), col=gray.colors(19, start = 0.9, end = 0.3))
dev.off()

#histogram of densities no log
pdf("histogram_normilized.pdf", h=10,w=10)
hist(cpm.matrix, col=gray.colors(19, start = 0.9, end = 0.3))
dev.off()

# Load library for pheatmap
cpm.matrix.corr <- cor(cpm.matrix, method="spearman",use="pairwise.complete.obs")

library("pheatmap")
library(gplots)
pdf(file = "sampleClusteringHeatmap.pdf", width = 50, height = 50);
pheatmap(cpm.matrix.corr, fontsize=50,cellwidth=50, cellheight=50, treeheight_col= 450, treeheight_row=450,angle_col=90, legend=F, cex.legend=1.2)
#heatmap.2(cpm.matrix.corr*10000,trace = "none",margins = c(5, 11),keysize =1 , key.title="",col ="bluered",density.info="none")
dev.off()

system("open sampleClusteringHeatmap.pdf")

system("open *pdf")

#write.table("cpm.matrix", file ="cpm.matrix.express.tab")

#write.table(cpm.matrix, file ="cpm.matrix.express.tab")


counts.mir166 <- sum(analysis_matrix$counts[grep(paste(c("miR166"),collapse="|"), rownames(analysis_matrix$counts)),])


counts.miRNA <- sum(analysis_matrix$counts[grep(paste(c("miR","Novel"),collapse="|"), rownames(analysis_matrix$counts)),])


counts.21PHAS <- sum(analysis_matrix$counts[grep(paste(c("21PHAS"),collapse="|"), rownames(analysis_matrix$counts)),])

counts.24PHAS <- sum(analysis_matrix$counts[grep(paste(c("24PHAS"),collapse="|"), rownames(analysis_matrix$counts)),])

counts.L24P <- sum(analysis_matrix$counts[grep(paste(c("L24P"),collapse="|"), rownames(analysis_matrix$counts)),])

counts.UNK <- sum(analysis_matrix$counts[grep(paste(c("UNK"),collapse="|"), rownames(analysis_matrix$counts)),])


counts.SNORNA <- sum(analysis_matrix$counts[grep(paste(c("SNORNA"),collapse="|"), rownames(analysis_matrix$counts)),])

counts.SNRNA <- sum(analysis_matrix$counts[grep(paste(c("SNRNA"),collapse="|"), rownames(analysis_matrix$counts)),])


counts.TRNA <- sum(analysis_matrix$counts[grep(paste(c("TRNA"),collapse="|"), rownames(analysis_matrix$counts)),])

mappedReads <-  c(counts.SNRNA,counts.SNORNA,counts.TRNA,counts.21PHAS,counts.24PHAS,counts.L24P,counts.UNK,counts.miRNA)

names(mappedReads) <- c("sn-RNA","sno-RNA","t-RNA","21-PHAS","24-PHAS","L24P-like","UNK","mi-RNA")


pastelColors <- c("#B97C7C",
"#698595",
"#9BA9CE",
"#C0C8CE",
"#CAB393",
"#BF7F58",
"#5F6D54",
"#BCDDE3"

)


pdf("mapedFragsments.pdf",h=5,w=7)

bp <- barplot(mappedReads,ylim=c(0,max(mappedReads)*1.20),yaxt='n',  col=pastelColors,xaxt="n")

mtext(side =1,at=bp,text= names(mappedReads),srt=-45, line=1,cex=0.8)

title(ylab="Millions of fragments",cex.lab=1.5,line=2.3,cex=0.7)

axis(2,seq(0,max(mappedReads)*1.2,50000000),labels=F)
mtext(side=2,text=c(seq(0,200,50)),outer=F,las=2,line=.8,at=seq(0,max(mappedReads)*1.2,50000000),cex=.5)


legend("topleft",
legend=names(mappedReads),
col = pastelColors,
#pch=21,
fill=pastelColors,
 bg = "gray95",
 pt.cex=2,
 border="black",
cex=.8)

dev.off()

system("open mapedFragsments.pdf")


#######################
#######CONTRASTS#######
#######################

#Node_x_B1

lrt_Node_x_B1 <- glmLRT(fit, contrast = c(1,0,0,0,0,0,0,-1))
tTags_lrt_Node_x_B1 <- topTags(lrt_Node_x_B1, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_B1$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_B1$table$logFC) >= 1)

sig_kept_Node_x_B1<- tTags_lrt_Node_x_B1$table[keep_sig[,1],]

sig_kept_Node_x_B1 <- sig_kept_Node_x_B1[order(sig_kept_Node_x_B1$logFC),]

write.table(sig_kept_Node_x_B1, file = "Node_x_B1.tab", row.names = T, quote = F, sep = "\t")

#Node_x_B2

lrt_Node_x_B2 <- glmLRT(fit, contrast = c(0,1,0,0,0,0,0,-1))
tTags_lrt_Node_x_B2 <- topTags(lrt_Node_x_B2, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_B2$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_B2$table$logFC) >= 1)

sig_kept_Node_x_B2<- tTags_lrt_Node_x_B2$table[keep_sig[,1],]

sig_kept_Node_x_B2 <- sig_kept_Node_x_B2[order(sig_kept_Node_x_B2$logFC),]

write.table(sig_kept_Node_x_B2, file = "Node_x_B2.tab", row.names = T, quote = F, sep = "\t")

#Node_x_B3

lrt_Node_x_B3 <- glmLRT(fit, contrast = c(0,0,1,0,0,0,0,-1))
tTags_lrt_Node_x_B3 <- topTags(lrt_Node_x_B3, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_B3$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_B3$table$logFC) >= 1)

sig_kept_Node_x_B3<- tTags_lrt_Node_x_B3$table[keep_sig[,1],]

sig_kept_Node_x_B3 <- sig_kept_Node_x_B3[order(sig_kept_Node_x_B3$logFC),]

write.table(sig_kept_Node_x_B3, file = "Node_x_B3.tab", row.names = T, quote = F, sep = "\t")

#Node_x_B4

lrt_Node_x_B4 <- glmLRT(fit, contrast = c(0,0,0,1,0,0,0,-1))
tTags_lrt_Node_x_B4 <- topTags(lrt_Node_x_B4, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_B4$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_B4$table$logFC) >= 1)

sig_kept_Node_x_B4<- tTags_lrt_Node_x_B4$table[keep_sig[,1],]

sig_kept_Node_x_B4 <- sig_kept_Node_x_B4[order(sig_kept_Node_x_B4$logFC),]

write.table(sig_kept_Node_x_B4, file = "Node_x_B4.tab", row.names = T, quote = F, sep = "\t")

#Node_x_B5

lrt_Node_x_B5 <- glmLRT(fit, contrast = c(0,0,0,0,1,0,0,-1))
tTags_lrt_Node_x_B5 <- topTags(lrt_Node_x_B5, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_B5$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_B5$table$logFC) >= 1)

sig_kept_Node_x_B5<- tTags_lrt_Node_x_B5$table[keep_sig[,1],]

sig_kept_Node_x_B5 <- sig_kept_Node_x_B5[order(sig_kept_Node_x_B5$logFC),]

write.table(sig_kept_Node_x_B5, file = "Node_x_B5.tab", row.names = T, quote = F, sep = "\t")

#Node_x_B6

lrt_Node_x_B6 <- glmLRT(fit, contrast = c(0,0,0,0,0,1,0,-1))
tTags_lrt_Node_x_B6 <- topTags(lrt_Node_x_B6, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_B6$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_B6$table$logFC) >= 1)

sig_kept_Node_x_B6<- tTags_lrt_Node_x_B6$table[keep_sig[,1],]

sig_kept_Node_x_B6 <- sig_kept_Node_x_B6[order(sig_kept_Node_x_B6$logFC),]

write.table(sig_kept_Node_x_B6, file = "Node_x_B6.tab", row.names = T, quote = F, sep = "\t")

#Node_x_Flower

lrt_Node_x_Flower <- glmLRT(fit, contrast = c(0,0,0,0,0,0,1,-1))
tTags_lrt_Node_x_Flower <- topTags(lrt_Node_x_Flower, n= NULL)
keep_sig <- matrix(tTags_lrt_Node_x_Flower$table$FDR <= 0.05 & abs(tTags_lrt_Node_x_Flower$table$logFC) >= 1)

sig_kept_Node_x_Flower<- tTags_lrt_Node_x_Flower$table[keep_sig[,1],]

sig_kept_Node_x_Flower <- sig_kept_Node_x_Flower[order(sig_kept_Node_x_Flower$logFC),]

write.table(sig_kept_Node_x_Flower, file = "Node_x_Flower.tab", row.names = T, quote = F, sep = "\t")

#B1_x_B2

lrt_B1_x_B2 <- glmLRT(fit, contrast = c(-1,1,0,0,0,0,0,0))
tTags_lrt_B1_x_B2 <- topTags(lrt_B1_x_B2, n= NULL)
keep_sig <- matrix(tTags_lrt_B1_x_B2$table$FDR <= 0.05 & abs(tTags_lrt_B1_x_B2$table$logFC) >= 1)

sig_kept_B1_x_B2<- tTags_lrt_B1_x_B2$table[keep_sig[,1],]

sig_kept_B1_x_B2 <- sig_kept_B1_x_B2[order(sig_kept_B1_x_B2$logFC),]

write.table(sig_kept_B1_x_B2, file = "B1_x_B2.tab", row.names = T, quote = F, sep = "\t")

#B2_x_B3

lrt_B2_x_B3 <- glmLRT(fit, contrast = c(0,-1,1,0,0,0,0,0))
tTags_lrt_B2_x_B3 <- topTags(lrt_B2_x_B3, n= NULL)
keep_sig <- matrix(tTags_lrt_B2_x_B3$table$FDR <= 0.05 & abs(tTags_lrt_B2_x_B3$table$logFC) >= 1)

sig_kept_B2_x_B3<- tTags_lrt_B2_x_B3$table[keep_sig[,1],]

sig_kept_B2_x_B3 <- sig_kept_B2_x_B3[order(sig_kept_B2_x_B3$logFC),]

write.table(sig_kept_B2_x_B3, file = "B2_x_B3.tab", row.names = T, quote = F, sep = "\t")

#B3_x_B4

lrt_B3_x_B4 <- glmLRT(fit, contrast = c(0,0,-1,1,0,0,0,0))
tTags_lrt_B3_x_B4 <- topTags(lrt_B3_x_B4, n= NULL)
keep_sig <- matrix(tTags_lrt_B3_x_B4$table$FDR <= 0.05 & abs(tTags_lrt_B3_x_B4$table$logFC) >= 1)

sig_kept_B3_x_B4<- tTags_lrt_B3_x_B4$table[keep_sig[,1],]

sig_kept_B3_x_B4 <- sig_kept_B3_x_B4[order(sig_kept_B3_x_B4$logFC),]

write.table(sig_kept_B3_x_B4, file = "B3_x_B4.tab", row.names = T, quote = F, sep = "\t")
#B4_x_B5

lrt_B4_x_B5 <- glmLRT(fit, contrast = c(0,0,0,-1,1,0,0,0))
tTags_lrt_B4_x_B5 <- topTags(lrt_B4_x_B5, n= NULL)
keep_sig <- matrix(tTags_lrt_B4_x_B5$table$FDR <= 0.05 & abs(tTags_lrt_B4_x_B5$table$logFC) >= 1)

sig_kept_B4_x_B5<- tTags_lrt_B4_x_B5$table[keep_sig[,1],]

sig_kept_B4_x_B5 <- sig_kept_B4_x_B5[order(sig_kept_B4_x_B5$logFC),]

write.table(sig_kept_B4_x_B5, file = "B4_x_B5.tab", row.names = T, quote = F, sep = "\t")

#B5_x_B6

lrt_B5_x_B6 <- glmLRT(fit, contrast = c(0,0,0,0,-1,1,0,0))
tTags_lrt_B5_x_B6 <- topTags(lrt_B5_x_B6, n= NULL)
keep_sig <- matrix(tTags_lrt_B5_x_B6$table$FDR <= 0.05 & abs(tTags_lrt_B5_x_B6$table$logFC) >= 1)

sig_kept_B5_x_B6<- tTags_lrt_B5_x_B6$table[keep_sig[,1],]

sig_kept_B5_x_B6 <- sig_kept_B5_x_B6[order(sig_kept_B5_x_B6$logFC),]

write.table(sig_kept_B5_x_B6, file = "B5_x_B6.tab", row.names = T, quote = F, sep = "\t")

#B6_x_Flower

lrt_B6_x_Flower <- glmLRT(fit, contrast = c(0,0,0,0,0,-1,1,0))
tTags_lrt_B6_x_Flower <- topTags(lrt_B6_x_Flower, n= NULL)
keep_sig <- matrix(tTags_lrt_B6_x_Flower$table$FDR <= 0.05 & abs(tTags_lrt_B6_x_Flower$table$logFC) >= 1)

sig_kept_B6_x_Flower<- tTags_lrt_B6_x_Flower$table[keep_sig[,1],]

sig_kept_B6_x_Flower <- sig_kept_B6_x_Flower[order(sig_kept_B6_x_Flower$logFC),]

write.table(sig_kept_B6_x_Flower, file = "B6_x_Flower.tab", row.names = T, quote = F, sep = "\t")


######################
###General DE Plots##
####################


dedupKeepedNames <-unique(c(rownames(sig_kept_B1_x_B2),
rownames(sig_kept_B2_x_B3),
rownames(sig_kept_B3_x_B4),
rownames(sig_kept_B4_x_B5),
rownames(sig_kept_B5_x_B6),
rownames(sig_kept_B6_x_Flower),
rownames(sig_kept_Node_x_B1),
rownames(sig_kept_Node_x_B2),
rownames(sig_kept_Node_x_B3),
rownames(sig_kept_Node_x_B4),
rownames(sig_kept_Node_x_B5),
rownames(sig_kept_Node_x_B6),
rownames(sig_kept_Node_x_Flower)))


sequencesLengths <- read.table("sequencesLengths.tab")

rownames(sequencesLengths) <- sequencesLengths$V1

sequencesLengths <- sequencesLengths[order(rownames(sequencesLengths)),]


sequencesLengths$V2 <- as.numeric(sequencesLengths$V2)

#RPKM

analysis_matrix$genes <- sequencesLengths
colnames(analysis_matrix$genes) <- c("names", "lengths")


DE.RPKM.matrix <- rpkm(analysis_matrix,normalized.lib.sizes=T)

DE.RPKM.matrix <- DE.RPKM.matrix[intersect(dedupKeepedNames, analysis_matrix$genes$names),]

cpm.matrix <- cpm(analysis_matrix,normalized.lib.sizes=T)



reorderColuns <- c( "Node_Cat_r1","Node_Cat_r2","Node_Cat_r3","Node_Sir_r1","Node_Sir_r2","Node_Sir_r3","B1_Cat_r2","B1_Cat_r3","B1_Sir_r1","B1_Sir_r2","B1_Sir_r3","B2_Cat_r1","B2_Cat_r2","B2_Cat_r3","B2_Sir_r1","B2_Sir_r2","B2_Sir_r3","B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3","B4_Cat_r1","B4_Cat_r2","B4_Cat_r3","B4_Sir_r1","B4_Sir_r2","B4_Sir_r3","B5_Cat_r1","B5_Cat_r2","B5_Cat_r3","B5_Sir_r1","B5_Sir_r2","B5_Sir_r3","B6_Cat_r1","B6_Cat_r2","B6_Cat_r3","B6_Sir_r1","B6_Sir_r2","B6_Sir_r3","Flower_Cat_r1","Flower_Cat_r2","Flower_Cat_r3","Flower_Sir_r1","Flower_Sir_r2","Flower_Sir_r3")

cpm.matrix <- cpm.matrix[,reorderColuns]
##########


#heatmap of all smallRNAs

newCol = colorRampPalette(c("white","#2975AF"))(100)
library(gplots)
pdf("heatmap_log_AllsmallExpressed.pdf", he=10,wi=10)
heatmap.2(log(cpm.matrix+1,10),
dendrogram = "row",
trace = "none",
key = T,
Colv=F,
Rowv=T,
margins = c(5, 2),
offsetRow =100,
key.xlab="Log 10 CPM + 1",
key.ylab="",
keysize =.9,
key.title="",
cexRow=1,
cexCol=0.8,
srtCol=45,
adjCol=c(1,1),
offsetCol = 1,
col=newCol,
denscol="black",
densadj = 5,
colsep=c(0:ncol(cpm.matrix)),
sepcolor="black",
sepwidth=c(0.01, 0.01)
)
dev.off()

system("open heatmap_log_AllsmallExpressed.pdf")


pastelColors <- c("#B97C7C",
"#698595",
"#9BA9CE",
"#C0C8CE",
"#CAB393",
"#BF7F58",
"#5F6D54",
"#BCDDE3"

)

################################
############SNRNA###############
################################

SnRna <- cpm.matrix[grep("SNRNA", rownames(cpm.matrix)),]


newCol = colorRampPalette(c("white",pastelColors[1]))(100)


pdf("heatmap_log_SNRNA.pdf", he=10,wi=10)
heatmap.2(log(SnRna+1,10),
dendrogram = "row",
trace = "none",
key = T,
Colv=F,
Rowv=T,
margins = c(5, 2),
offsetRow =100,
key.xlab="Log 10 CPM + 1",
key.ylab="",
keysize =.9,
key.title="",
cexRow=1,
cexCol=0.8,
srtCol=45,
adjCol=c(1,1),
offsetCol = 1,
col=newCol,
denscol="black",
densadj = 5,
colsep=c(0:ncol(cpm.matrix)),
sepcolor="black",
sepwidth=c(0.01, 0.01)
)
dev.off()

system("open heatmap_log_SNRNA.pdf")



#################################
############SNORNA###############
#################################


  SnoRna <- cpm.matrix[grep("SNORNA", rownames(cpm.matrix)),]


  newCol = colorRampPalette(c("white",pastelColors[2]))(100)


  pdf("heatmap_log_SNORNA.pdf", he=10,wi=10)
  heatmap.2(log(SnoRna+1,10),
  dendrogram = "row",
  trace = "none",
  key = T,
  Colv=F,
  Rowv=T,
  margins = c(5, 2),
  offsetRow =100,
  key.xlab="Log 10 CPM + 1",
  key.ylab="",
  keysize =.9,
  key.title="",
  cexRow=1,
  cexCol=0.8,
  srtCol=45,
  adjCol=c(1,1),
  offsetCol = 1,
  col=newCol,
  denscol="black",
  densadj = 5,
  colsep=c(0:ncol(cpm.matrix)),
  sepcolor="black",
  sepwidth=c(0.01, 0.01)
  )
  dev.off()

  system("open heatmap_log_SNORNA.pdf")


  #################################
  ##############TRNA###############
  #################################


    tRna <- cpm.matrix[grep("TRNA", rownames(cpm.matrix)),]


    newCol = colorRampPalette(c("white",pastelColors[3]))(100)


    pdf("heatmap_log_TRNA.pdf", he=10,wi=10)
    heatmap.2(log(tRna+1,10),
    dendrogram = "row",
    trace = "none",
    key = T,
    Colv=F,
    Rowv=T,
    margins = c(5, 2),
    offsetRow =100,
    key.xlab="Log 10 CPM + 1",
    key.ylab="",
    keysize =.9,
    key.title="",
    cexRow=1,
    cexCol=0.8,
    srtCol=45,
    adjCol=c(1,1),
    offsetCol = 1,
    col=newCol,
    denscol="black",
    densadj = 5,
    colsep=c(0:ncol(cpm.matrix)),
    sepcolor="black",
    sepwidth=c(0.01, 0.01)
    )
    dev.off()

    system("open heatmap_log_TRNA.pdf")


#################################
##############21PHAS###############
#################################


PHAS21Rna <- cpm.matrix[grep("21PHAS", rownames(cpm.matrix)),]


newCol = colorRampPalette(c("white",pastelColors[4]))(100)


      pdf("heatmap_log_21PHAS.pdf", he=10,wi=10)
      heatmap.2(log(PHAS21Rna+1,10),
      dendrogram = "row",
      trace = "none",
      key = T,
      Colv=F,
      Rowv=T,
      margins = c(5, 2),
      offsetRow =100,
      key.xlab="Log 10 CPM + 1",
      key.ylab="",
      keysize =.9,
      key.title="",
      cexRow=1,
      cexCol=0.8,
      srtCol=45,
      adjCol=c(1,1),
      offsetCol = 1,
      col=newCol,
      denscol="black",
      densadj = 5,
      colsep=c(0:ncol(cpm.matrix)),
      sepcolor="black",
      sepwidth=c(0.01, 0.01)
      )
      dev.off()

      system("open heatmap_log_21PHAS.pdf")

#################################
##############24PHAS#############
#################################


PHAS24Rna <- cpm.matrix[grep("24PHAS", rownames(cpm.matrix)),]


newCol = colorRampPalette(c("white",pastelColors[5]))(100)


        pdf("heatmap_log_24PHAS.pdf", he=10,wi=10)
        heatmap.2(log(PHAS24Rna+1,10),
        dendrogram = "row",
        trace = "none",
        key = T,
        Colv=F,
        Rowv=T,
        margins = c(5, 2),
        offsetRow =100,
        key.xlab="Log 10 CPM + 1",
        key.ylab="",
        keysize =.9,
        key.title="",
        cexRow=1,
        cexCol=0.8,
        srtCol=45,
        adjCol=c(1,1),
        offsetCol = 1,
        col=newCol,
        denscol="black",
        densadj = 5,
        colsep=c(0:ncol(cpm.matrix)),
        sepcolor="black",
        sepwidth=c(0.01, 0.01)
        )
        dev.off()

        system("open heatmap_log_24PHAS.pdf")

#################################
#########L24P.like###############
#################################


L24P.likeRna <- cpm.matrix[grep("L24P.like", rownames(cpm.matrix)),]


newCol = colorRampPalette(c("white",pastelColors[6]))(100)


                pdf("heatmap_log_L24P.like.pdf", he=10,wi=10)
                heatmap.2(log(L24P.likeRna+1,10),
                dendrogram = "row",
                trace = "none",
                key = T,
                Colv=F,
                Rowv=T,
                margins = c(5, 2),
                offsetRow =100,
                key.xlab="Log 10 CPM + 1",
                key.ylab="",
                keysize =.9,
                key.title="",
                cexRow=1,
                cexCol=0.8,
                srtCol=45,
                adjCol=c(1,1),
                offsetCol = 1,
                col=newCol,
                denscol="black",
                densadj = 5,
                colsep=c(0:ncol(cpm.matrix)),
                sepcolor="black",
                sepwidth=c(0.01, 0.01)
                )
                dev.off()

                system("open heatmap_log_L24P.like.pdf")

#################################
###############UNK###############
#################################


UNKRna <- cpm.matrix[grep("UNK", rownames(cpm.matrix)),]


newCol = colorRampPalette(c("white",pastelColors[7]))(100)


pdf("heatmap_log_UNK.like.pdf", he=10,wi=10)
heatmap.2(log(UNKRna+1,10),
dendrogram = "row",
trace = "none",
key = T,
Colv=F,
Rowv=T,
margins = c(5, 2),
offsetRow =100,
key.xlab="Log 10 CPM + 1",
key.ylab="",
keysize =.9,
key.title="",
cexRow=1,
cexCol=0.8,
srtCol=45,
adjCol=c(1,1),
offsetCol = 1,
col=newCol,
denscol="black",
densadj = 5,
colsep=c(0:ncol(cpm.matrix)),
sepcolor="black",
sepwidth=c(0.01, 0.01)
)
dev.off()

system("open heatmap_log_UNK.like.pdf")


#################################
###############UNK###############
#################################


MIRna <- cpm.matrix[grep(paste(c("miR","Novel"),collapse="|"), rownames(cpm.matrix)),]


newCol = colorRampPalette(c("white",pastelColors[8]))(100)


pdf("heatmap_log_MIR.pdf", he=10,wi=10)
heatmap.2(log(MIRna+1,10),
dendrogram = "row",
trace = "none",
key = T,
Colv=F,
Rowv=T,
margins = c(5, 2),
offsetRow =100,
key.xlab="Log 10 CPM + 1",
key.ylab="",
keysize =.9,
key.title="",
cexRow=1,
cexCol=0.8,
srtCol=45,
adjCol=c(1,1),
offsetCol = 1,
col=newCol,
denscol="black",
densadj = 5,
colsep=c(0:ncol(cpm.matrix)),
sepcolor="black",
sepwidth=c(0.01, 0.01)
)
dev.off()

system("open heatmap_log_MIR.pdf")

##########

#Check if in B3 the L24P-like are higlly expressed when compared to the other types - this can suggest that they are some type of sirenRNA

L24P.likeB3<- L24P.likeRna[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]

PHAS24Rna.likeB3<- PHAS24Rna[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]

UNKRnaB3 <- UNKRna[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]

SnRnaB3 <- SnRna[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]

SnoRnaB3 <- SnoRna[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]

tRnaB3 <- tRna[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]

miRnaB3 <- MIRna[rownames(MIRna) != "miR166.1.ab",c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]


cpm.matrix.B3 <- cpm.matrix[,c("B3_Cat_r1","B3_Cat_r2","B3_Cat_r3","B3_Sir_r1","B3_Sir_r2","B3_Sir_r3")]


##########

DE.cpm.matrix <- cpm.matrix[intersect(dedupKeepedNames, rownames(analysis_matrix$counts)),]

means.Flower <- as.matrix(rowMeans(cpm.matrix[,c(1:6)]))
means.Node <- as.matrix(rowMeans(cpm.matrix[,c(7:12)]))
means.B1 <- as.matrix(rowMeans(cpm.matrix[,c(13:17)]))
means.B2 <- as.matrix(rowMeans(cpm.matrix[,c(18:23)]))
means.B3 <- as.matrix(rowMeans(cpm.matrix[,c(24:26,30:32)]))

means.B4 <- as.matrix(rowMeans(cpm.matrix[,c(27:29,33:35)]))
means.B5 <- as.matrix(rowMeans(cpm.matrix[,c(36:41)]))

means.B6 <- as.matrix(rowMeans(cpm.matrix[,c(42:47)]))

per_condition_means <- cbind(means.Node,means.B1,means.B2, means.B3,means.B4,means.B5,means.B6,means.Flower)

colnames(per_condition_means) <- c("Node","B1","B2","B3","B4","B5","B6","Flower")


#blue.red <- colorRampPalette(c("#B6D7B3","#d93e23"))(5)

#"#7bc437"
#
#blue.red <- colorRampPalette(c("#7bc437","#c24a3a"))(5)

blue.red <- colorRampPalette(c("white",cols1[47]))(10)

library(gplots)
pdf("heatmap_log_absolute_Expression.pdf", he=10,wi=10)
heatmap.2(log(per_condition_means+1,10),dendrogram = "both",trace = "none",key = T, Colv=T,Rowv=T,margins = c(5, 2),offsetRow =100, key.xlab="Log 10 CPM + 1", key.ylab="", keysize =.9 , key.title="", cexRow=1, cexCol=1.2,srtCol=0,adjCol=c(0.5,1.5), offsetCol = .3, col=blue.red, denscol="blue", densadj = 5,colsep=c(0:ncol(per_condition_means)),sepcolor="black",sepwidth=c(0.01, 0.01))
dev.off()

system("open heatmap_log_absolute_Expression.pdf")



#all DE genes barplot
#####REMEBER TO CHANGE TO THE cpm.matrix AND NOT THE DE.RPKM MATRIX!!!
sd.Node <-as.matrix(apply(cpm.matrix[,c(7:12)],1,sd))
sd.B1<-as.matrix(apply(cpm.matrix[,c(13:17)],1,sd))
sd.B2<-as.matrix(apply(cpm.matrix[,c(18:23)],1,sd))
sd.B3<-as.matrix(apply(cpm.matrix[,c(24:26,30:32)],1,sd))
sd.B4<-as.matrix(apply(cpm.matrix[,c(27:29,33:35)],1,sd))
sd.B5<-as.matrix(apply(cpm.matrix[,c(36:41)],1,sd))
sd.B6<-as.matrix(apply(cpm.matrix[,c(42:47)],1,sd))
sd.Flower<-as.matrix(apply(cpm.matrix[,c(1:6)],1,sd))


per_condition_sd <- cbind(sd.Node,sd.B1,sd.B2, sd.B3,sd.B4,sd.B5,sd.B6,sd.Flower)

colnames(per_condition_sd) <- c("Node","B1","B2","B3","B4","B5","B6","Flower")

#"#7bc437"
green.red <- colorRampPalette(c("#5F6D54","#CAB393","#BF7F58"))(8)
#"#c24a3a","#522c28"
i=10
system("mkdir barplot")
for (gene in rownames(per_condition_means)){
print(gene)
limitOfY <- max(per_condition_means[gene,]+per_condition_sd[gene,]/sqrt(6))*1.2
pdf(paste0("./barplot/barplot",gene,".pdf"),h=10,w=17)
barplot <- barplot(per_condition_means[gene,],beside=T,col=green.red, yaxt='n',xaxt='n',ylim=c(0,limitOfY),main =paste("Expression of",gene),cex.main=2.3,width=1.2,space=.9)

#space=c(.2,0,0,0,.2,0,0,0,.2,0,0,0,.2,0,0,0)

title(ylab="CPM",cex.lab=1.5,line=2.3)

#legend("topleft",legend=c("Node","B1", "B2","B3","B4","B5","B6", "Flower"),col = green.red,pch=19,cex=1.5)

if( limitOfY <= 50 ){
i=3
}
if( limitOfY > 50 & limitOfY <= 100 ){
i=10
}
if( limitOfY > 100 & limitOfY <= 1000 ){
i=50
}
if( limitOfY > 1000 & limitOfY <= 10000 ){
i=100
}
if( limitOfY > 10000 & limitOfY <= 100000 ){
i=1000
}
if( limitOfY > 100000 & limitOfY <= 1000000 ){
i=10000
}
if( limitOfY > 1000000 & limitOfY <= 10000000 ){
i=100000
}

axis(2,seq(0,limitOfY,i),labels=F)
mtext(side=2,text=seq(0,limitOfY,i),outer=F,las=2,line=.8,at=seq(0,limitOfY,i),cex=.8)


mtext(side=1,text=c("Node","B1","B2","B3","B4","B5","B6","Flower"),outer=F,line=1.2,at=barplot,cex=2)


arrows(x0 = barplot, y0 = per_condition_means[gene,] - per_condition_sd[gene,]/sqrt(5), x1 = barplot, y1=per_condition_means[gene,] + per_condition_sd[gene,]/sqrt(5) ,code=3,angle=90,length=0.05,col="darkgreen",lwd=4)

dev.off()

}

write.table(rnaseqmatrix,file="rnaseq.matrix.tab",sep="\t")


#multiple volcano plots
###########################################

system("mkdir volcanoes")
l <- ls(pattern="tTags_lrt_")

for (comp in l){
print(comp)
print(max(get(comp)$table$logFC))
print(min(get(comp)$table$logFC))
}

for (comp in l){
print(comp)
print(max(-log(get(comp)$table$FDR,10)))
}



#Volcano plots
for (comp in l){
print(comp)
FC <-  get(comp)$table$logFC
names(FC) <- rownames(get(comp)$table)

FDR <- get(comp)$table$FDR
DATA <- cbind(FC,FDR)

pdf(paste0("./volcanoes/",comp,".pdf"),h=5,w=5)
par(mar = c(4.1, 4.1, 4.1, 2.1))

plot(-log(DATA[,2],10) ~ DATA[,1],xlim=c(-10,10),ylim=c(0,100),cex=.1,col="grey",yaxt='n',xaxt='n',ylab="",xlab="")

title(main=gsub("tTags_lrt_","" ,comp),ylab="-log10(FDR)",cex.lab=.9,line=2.3,xlab="log2FC")

legend("topleft",legend=c("Up-regulated","Not DE","Down-regulated"),col = c("#d93e23","grey","#236cd9" ),pch=19,cex=.6)


axis(2,seq(0,100,5),labels=F)
mtext(side=2,text=seq(0,100,5),outer=F,las=2,line=.8,at=seq(0,100,5),cex=.8)

text(x=-7.5,y=-log(0.05,10) +3.5,labels="FDR = 0.05",las=2, col = "#0e4f34",cex=.9)

axis(1,seq(-10,10,2),labels=F)
mtext(side=1,text=seq(-10,10,2),outer=F,las=1,line=.8,at=seq(-10,10,2),cex=.8)

axis(1,0,labels=F)
mtext(side=1,text=0,outer=F,las=1,line=.8,at=0,cex=.8)

points(-log(DATA[,2][DATA[,2] < 0.05 & DATA[,1] >= 1],10) ~DATA[,1][DATA[,2] < 0.05 & DATA[,1] >= 1], col = "#d93e23",cex=.1)

points(-log(DATA[,2][DATA[,2] < 0.05 & DATA[,1] <= -1],10) ~DATA[,1][DATA[,2] < 0.05 & DATA[,1] <= -1], col = "#236cd9",cex=.1)

abline(v= 1, col = "#0e4f34",lty = 2)
abline(v= -1, col = "#0e4f34",lty = 2)
abline(h= -log(0.05,10), col = "#0e4f34",lty = 2)

dev.off()
}



for (comp in l){
print(comp)
print(max(get(comp)$table$logFC))
print(min(get(comp)$table$logFC))
}


for (comp in l){
print(comp)
print(max(get(comp)$table$logCPM))
}

for (comp in l){
print(comp)
print(max(-log(get(comp)$table$FDR,10)))
}

#MA plots
#INSIDE
system("mkdir MA")

for (comp in l){
print(comp)
FC <-  get(comp)$table$logFC
names(FC) <- rownames(get(comp)$table)

FDR <- get(comp)$table$FDR
lCPM <- get(comp)$table$logCPM
DATA <- cbind(FC,lCPM,FDR)

pdf(paste0("./MA/",comp,".pdf"),h=5,w=5)
par(mar = c(4.1, 4.1, 4.1, 2.1))

plot(DATA[,1] ~ DATA[,2] ,xlim=c(0,18),ylim=c(-9,9),cex=.1,col="grey",yaxt='n',xaxt='n',ylab="",xlab="")

title(main=gsub("tTags_lrt_","" ,comp),ylab="log2FC",cex.lab=.9,line=2.3,xlab="logCounts (CPM)")

#legend("topleft",legend=c("Up-regulated","Not DE","Down-regulated"),col = c("#d93e23","grey","#236cd9" ),pch=19,cex=.6)

axis(1,seq(0,18,2),labels=F)
mtext(side=1,text=seq(0,18,2),outer=F,las=1,line=.8,at=seq(0,18,2),cex=.8)

#text(x=-8.5,y=-log(0.05,10) +1.5,labels="FDR = 0.05",las=2, col = "#0e4f34",cex=.9)

axis(2,seq(-9,9,1),labels=F)
mtext(side=2,text=seq(-9,9,1),outer=F,las=1,line=.8,at=seq(-9,9,1),cex=.8)

#axis(1,0,labels=F)
#mtext(side=1,text=0,outer=F,las=1,line=.8,at=0,cex=.8)

points(DATA[,1][DATA[,3] < 0.05 & DATA[,1] >= 1] ~DATA[,2][DATA[,3] < 0.05 & DATA[,1] >= 1], col = "#d93e23",cex=.1)

points(DATA[,1][DATA[,3] < 0.05 & DATA[,1] <= -1] ~DATA[,2][DATA[,3] < 0.05 & DATA[,1] <= -1], col = "#236cd9",cex=.1)

#abline(v= 1, col = "#0e4f34",lty = 2)
#abline(v= -1, col = "#0e4f34",lty = 2)
#abline(h= -log(0.05,10), col = "#0e4f34",lty = 2)

#lightBlue "#8ce4ff"

#DOWN NAMES
#text(DATA[DownInside,1] ~ DATA[DownInside,2],labels= DownInside,cex=.25,pos=1)

dev.off()
}


##########################################
#Small RNAs aligned per chromossome

######
#mappedChr_1c <- rnaseqmatrix[grep("NC_039898.1",rownames(rnaseqmatrix)),]


classExpressionMatrix <- matrix()

#############################
############miRNA############
#############################

miRNAoverall <- colMeans(log(DE.RPKM.matrix[grep(paste(c("miR","Novel"),collapse="|"),rownames(analysis_matrix$counts)),] + 1))


#PHAS
PHASoverall <- colMeans(log(DE.RPKM.matrix[grep("PHAS",rownames(DE.RPKM.matrix)),]+1))

#SNORNA
SNORNAoverall <- colMeans(log(DE.RPKM.matrix[grep("SNORNA",rownames(DE.RPKM.matrix)),]+1))

#SNRNA
SNRNAoverall <- colMeans(log(DE.RPKM.matrix[grep("SNRNA",rownames(DE.RPKM.matrix)),]+1))


#TRNA
TRNAoverall <- colMeans(log(DE.RPKM.matrix[grep("TRNA",rownames(DE.RPKM.matrix)),]+1))

classExpressionMatrix <- cbind(miRNAoverall, PHASoverall,SNORNAoverall, SNRNAoverall,TRNAoverall )

colnames(classExpressionMatrix) <- c("miRNA", "phasRNA", "snoRNA","snRNA","tRNA")


t=20

pdf(file = "sMallOverallExpression.pdf", width = 10, height = 10);
par(mfrow=c(1,3))
pheatmap(classExpressionMatrix, fontsize=10,cellwidth=10, cellheight=10,angle_col=90, legend=T, cex.legend=1.2,cluster_rows = FALSE,main="Small RNAs overall expression")
dev.off()

system("open sMallOverallExpression.pdf")

save.image("DEsmallRNAseq.RData")


###############################################
#Create stacked barplots with the number of DE genes in each comparisson to Node
###############################################

#load the data regarding 24 or 21 phas loci
#phasing <- read.delim("shortStackResults/phasing.tab")



comparissonsOfInterest <- c(
"sig_kept_Node_x_Flower",
"sig_kept_Node_x_B6",
"sig_kept_Node_x_B5",
"sig_kept_Node_x_B4",
"sig_kept_Node_x_B3",
"sig_kept_Node_x_B2",
"sig_kept_Node_x_B1"

)

comparissonsOfInterestExpressedUP <- data.frame()

comparissonsOfInterestExpressedDOWN <- data.frame()

for (comparisson in comparissonsOfInterest){
  print(comparisson)

  namesUP <- rownames(get(comparisson)[get(comparisson)$logFC > 0,])

  countSnRnaUP <- length(grep("SNRNA",namesUP))
  countSnoRnaUP <- length(grep("SNORNA",namesUP))
  countTRnaUP <- length(grep("TRNA",namesUP))
  countPHAS21UP <- length(grep("21PHAS",namesUP))
  countPHAS24UP <- length(grep("24PHAS",namesUP))
  countL24P.likeUP <- length(grep("L24P.like",namesUP))
  countUNKUP <- length(grep("UNK",namesUP))

  countMirRnaUP <- length(grep(paste(c("miR","Novel"),collapse="|"),namesUP))

  assign(paste("miRUP",comparisson,sep=""),namesUP[grep(paste(c("miR","Novel"),collapse="|"),namesUP)])

  countUP <- c(countSnRnaUP,countSnoRnaUP,countTRnaUP,countPHAS21UP,countPHAS24UP,countL24P.likeUP,countUNKUP,countMirRnaUP)

  namesDOWN <- rownames(get(comparisson)[get(comparisson)$logFC < 0,])

  countSnRnaDOWN <- length(grep("SNRNA",namesDOWN))
  countSnoRnaDOWN <- length(grep("SNORNA",namesDOWN))
  countTRnaDOWN <- length(grep("TRNA",namesDOWN))
  countPHAS21DOWN <- length(grep("21PHAS",namesDOWN))
  countPHAS24DOWN <- length(grep("24PHAS",namesDOWN))
  countL24P.likeDOWN <- length(grep("L24P.like",namesDOWN))
  countUNK <- length(grep("UNK",namesDOWN))


  countMirRnaDOWN <- length(grep(paste(c("miR","Novel"),collapse="|"),namesDOWN))

  assign(paste("miRDOWN",comparisson,sep=""),namesDOWN[grep(paste(c("miR","Novel"),collapse="|"),namesDOWN)])


  countDOWN <- c(countSnRnaDOWN,countSnoRnaDOWN,countTRnaDOWN,countPHAS21DOWN,countPHAS24DOWN,countL24P.likeDOWN,countUNK,countMirRnaDOWN)

  print(countDOWN)
  print(countUP)

  if(nrow(comparissonsOfInterestExpressedDOWN) == 0){

      comparissonsOfInterestExpressedDOWN <- as.data.frame(countDOWN)

  }else{
    comparissonsOfInterestExpressedDOWN <- cbind(comparissonsOfInterestExpressedDOWN,  countDOWN)
  }

if(nrow(comparissonsOfInterestExpressedUP) == 0){
  comparissonsOfInterestExpressedUP <- as.data.frame(countUP)

}else{

  comparissonsOfInterestExpressedUP <- cbind(comparissonsOfInterestExpressedUP,
countUP)
  }

}

colnames(comparissonsOfInterestExpressedDOWN) <- c(
"Node_x_Flower",
"Node_x_B6",
"Node_x_B5",
"Node_x_B4",
"Node_x_B3",
"Node_x_B2",
"Node_x_B1"
)

#vempraca
rownames(comparissonsOfInterestExpressedDOWN)<- c("sn-RNA","sno-RNA","t-RNA","21-PHAS","24-PHAS","L24P-like","UNK","mi-RNA")


colnames(comparissonsOfInterestExpressedUP)<- c(
"Node_x_Flower",
"Node_x_B6",
"Node_x_B5",
"Node_x_B4",
"Node_x_B3",
"Node_x_B2",
"Node_x_B1"
)
#MUDAR AQUI PELAS CORES
rownames(comparissonsOfInterestExpressedUP)<- c("sn-RNA","sno-RNA","t-RNA","21-PHAS","24-PHAS","L24P-like","UNK","mi-RNA")

pastelColors <- c("#B97C7C",
"#698595",
"#9BA9CE",
"#C0C8CE",
"#CAB393",
"#BF7F58",
"#5F6D54",
"#BCDDE3"

)

pdf("Node_x_stages.pdf", h= 7,w=7)

barplot <- barplot(as.matrix(comparissonsOfInterestExpressedUP),horiz=TRUE,col=pastelColors, xlim=c(-500,600),yaxt='n',xaxt='n',
space=.6
)

barplot(as.matrix(-comparissonsOfInterestExpressedDOWN),horiz=TRUE,col=pastelColors,add=T,yaxt='n',xaxt='n',
space=.6
)

title(xlab="Number of DE transcripts",cex.lab=1,line=3)

mtext(side=2,text=colnames(comparissonsOfInterestExpressedUP),outer=F,line=-3,at=barplot,cex=1.1,las=2)

legend("topright",
legend=rownames(comparissonsOfInterestExpressedDOWN),
col = pastelColors,
#pch=21,
fill=pastelColors,
 bg = "gray95",
 pt.cex=2,
 border="black",
cex=.8)

abline(v=0,lwd=2,col=adjustcolor("black", alpha = 0.6))


axis(1,seq(-500,600,50),labels=F)
mtext(1,text=c(seq(500,0,-50),seq(50,600,50)),outer=F,las=1,line=.8,at=seq(-500,600,50),cex=.7)

axis(3,c(-500,0),labels=F)
mtext(3, text="UP regulated in Node",las=1,line=.1,at =-250,cex=1.2)


dev.off()

system("open Node_x_stages.pdf")

setequal(el=c(miRDOWNsig_kept_Node_x_B2,
miRDOWNsig_kept_Node_x_B3,
miRDOWNsig_kept_Node_x_B4,
miRDOWNsig_kept_Node_x_B2,
miRDOWNsig_kept_Node_x_B6,
miRDOWNsig_kept_Node_x_Flower))


unique(miRDOWNsig_kept_Node_x_B2,
miRDOWNsig_kept_Node_x_B3,
miRDOWNsig_kept_Node_x_B4,
miRDOWNsig_kept_Node_x_B2,
miRDOWNsig_kept_Node_x_B6,
miRDOWNsig_kept_Node_x_Flower)

Reduce(intersect, list(miRDOWNsig_kept_Node_x_B2,
miRDOWNsig_kept_Node_x_B3,
miRDOWNsig_kept_Node_x_B4,
miRDOWNsig_kept_Node_x_B2,
miRDOWNsig_kept_Node_x_B6,
miRDOWNsig_kept_Node_x_Flower))


Reduce(intersect, list(miRUPsig_kept_Node_x_B2,
miRUPsig_kept_Node_x_B3,
miRUPsig_kept_Node_x_B4,
miRUPsig_kept_Node_x_B2,
miRUPsig_kept_Node_x_B6,
miRUPsig_kept_Node_x_Flower))


###############################################
#Create stacked barplots with the number of DE genes in each comparisson to the next one
###############################################

comparissonsOfInterest <- c(
"sig_kept_B6_x_Flower",
"sig_kept_B5_x_B6",
"sig_kept_B4_x_B5",
"sig_kept_B3_x_B4",
"sig_kept_B2_x_B3",
"sig_kept_B1_x_B2",
"sig_kept_Node_x_B1"
)

comparissonsOfInterestExpressedUP <- data.frame()

comparissonsOfInterestExpressedDOWN <- data.frame()

for (comparisson in comparissonsOfInterest){
print(comparisson)

namesUP <- rownames(get(comparisson)[get(comparisson)$logFC > 0,])

countSnRnaUP <- length(grep("SNRNA",namesUP))
countSnoRnaUP <- length(grep("SNORNA",namesUP))
countTRnaUP <- length(grep("TRNA",namesUP))
countPHAS21UP <- length(grep("21PHAS",namesUP))
countPHAS24UP <- length(grep("24PHAS",namesUP))
countL24P.likeUP <- length(grep("L24P.like",namesUP))
countUNKUP <- length(grep("UNK",namesUP))

countMirRnaUP <- length(grep(paste(c("miR","Novel"),collapse="|"),namesUP))

countUP <- c(countSnRnaUP,countSnoRnaUP,countTRnaUP,countPHAS21UP,countPHAS24UP,countL24P.likeUP,countUNKUP,countMirRnaUP)

namesDOWN <- rownames(get(comparisson)[get(comparisson)$logFC < 0,])

countSnRnaDOWN <- length(grep("SNRNA",namesDOWN))
countSnoRnaDOWN <- length(grep("SNORNA",namesDOWN))
countTRnaDOWN <- length(grep("TRNA",namesDOWN))
countPHAS21DOWN <- length(grep("21PHAS",namesDOWN))
countPHAS24DOWN <- length(grep("24PHAS",namesDOWN))
countL24P.likeDOWN <- length(grep("L24P.like",namesDOWN))
countUNK <- length(grep("UNK",namesDOWN))


countMirRnaDOWN <- length(grep(paste(c("miR","Novel"),collapse="|"),namesDOWN))

countDOWN <- c(countSnRnaDOWN,countSnoRnaDOWN,countTRnaDOWN,countPHAS21DOWN,countPHAS24DOWN,countL24P.likeDOWN,countUNK,countMirRnaDOWN)

print(countDOWN)
print(countUP)

if(nrow(comparissonsOfInterestExpressedDOWN) == 0){

comparissonsOfInterestExpressedDOWN <- as.data.frame(countDOWN)

}else{
  comparissonsOfInterestExpressedDOWN <- cbind(comparissonsOfInterestExpressedDOWN,  countDOWN)
}

if(nrow(comparissonsOfInterestExpressedUP) == 0){
 comparissonsOfInterestExpressedUP <- as.data.frame(countUP)

}else{

  comparissonsOfInterestExpressedUP <- cbind(comparissonsOfInterestExpressedUP,
countUP)
}

}

colnames(comparissonsOfInterestExpressedDOWN) <- c(
"B6_x_Flower",
"B5_x_B6",
"B4_x_B5",
"B3_x_B4",
"B2_x_B3",
"B1_x_B2",
"Node_x_B1"
)

rownames(comparissonsOfInterestExpressedDOWN)<- c("sn-RNA","sno-RNA","t-RNA","21-PHAS","24-PHAS","L24P-like","UNK","mi-RNA")


colnames(comparissonsOfInterestExpressedUP)  <- c(
"B6_x_Flower",
"B5_x_B6",
"B4_x_B5",
"B3_x_B4",
"B2_x_B3",
"B1_x_B2",
"Node_x_B1"
)

rownames(comparissonsOfInterestExpressedUP)<- c("sn-RNA","sno-RNA","t-RNA","21-PHAS","24-PHAS","L24P-like","UNK","mi-RNA")

pastelColors <- c("#B97C7C",
"#698595",
"#9BA9CE",
"#C0C8CE",
"#CAB393",
"#BF7F58",
"#5F6D54",
"#BCDDE3"

)

pdf("stages_x_stages.pdf", h= 7,w=7)

barplot <- barplot(as.matrix(comparissonsOfInterestExpressedUP),horiz=TRUE,col=pastelColors, xlim=c(-500,600),yaxt='n',xaxt='n',
space=.6
)

barplot(as.matrix(-comparissonsOfInterestExpressedDOWN),horiz=TRUE,col=pastelColors,add=T,yaxt='n',xaxt='n',
space=.6
)

title(xlab="Number of DE transcripts",cex.lab=1,line=3)

#mtext(side=2,text=c(),outer=F,line=-3,at=barplot,cex=1.1,las=2)

legend("topleft",
legend=rownames(comparissonsOfInterestExpressedDOWN),
col = pastelColors,
#pch=21,
fill=pastelColors,
bg = "gray95",
pt.cex=2,
border="black",
cex=0.8)

abline(v=0,lwd=2,col=adjustcolor("black", alpha = 0.6))

axis(1,seq(-500,600,50),labels=F)
mtext(1,text=c(seq(500,0,-50),seq(50,600,50)),outer=F,las=1,line=.8,at=seq(-500,600,50),cex=.7)

#axis(3,c(-500,0),labels=F)
#mtext(3, text="UP regulated in Node",las=1,line=.1,at =-250,cex=1.2)

#left side

dev.off()

system("open stages_x_stages.pdf")


cpm.matrix <- cpm(analysis_matrix,normalized.lib.sizes=T)


#cpm.matrix <- cpm.matrix[intersect(dedupKeepedNames, rownames(analysis_matrix$counts)),]

cpm.means.Flower <- as.matrix(rowMeans(cpm.matrix[,c(1:6)]))
cpm.means.Node <- as.matrix(rowMeans(cpm.matrix[,c(7:12)]))
cpm.means.B1 <- as.matrix(rowMeans(cpm.matrix[,c(13:17)]))
cpm.means.B2 <- as.matrix(rowMeans(cpm.matrix[,c(18:23)]))
cpm.means.B3 <- as.matrix(rowMeans(cpm.matrix[,c(24:26,30:32)]))

cpm.means.B4 <- as.matrix(rowMeans(cpm.matrix[,c(27:29,33:35)]))
cpm.means.B5 <- as.matrix(rowMeans(cpm.matrix[,c(36:41)]))

cpm.means.B6 <- as.matrix(rowMeans(cpm.matrix[,c(42:47)]))

per_condition_means <- cbind(cpm.means.Node,cpm.means.B1,cpm.means.B2, cpm.means.B3,cpm.means.B4,cpm.means.B5,cpm.means.B6,cpm.means.Flower)

colnames(per_condition_means) <- c("Node","B1","B2","B3","B4","B5","B6","Flower")


#all genes barplot
#####REMEBER TO CHANGE TO THE cpm.matrix AND NOT THE DE.RPKM MATRIX!!!
sd.Node <-as.matrix(apply(cpm.matrix[,c(7:12)],1,sd))
sd.B1<-as.matrix(apply(cpm.matrix[,c(13:17)],1,sd))
sd.B2<-as.matrix(apply(cpm.matrix[,c(18:23)],1,sd))
sd.B3<-as.matrix(apply(cpm.matrix[,c(24:26,30:32)],1,sd))
sd.B4<-as.matrix(apply(cpm.matrix[,c(27:29,33:35)],1,sd))
sd.B5<-as.matrix(apply(cpm.matrix[,c(36:41)],1,sd))
sd.B6<-as.matrix(apply(cpm.matrix[,c(42:47)],1,sd))
sd.Flower<-as.matrix(apply(cpm.matrix[,c(1:6)],1,sd))


per_condition_sd <- cbind(sd.Node,sd.B1,sd.B2, sd.B3,sd.B4,sd.B5,sd.B6,sd.Flower)

colnames(per_condition_sd) <- c("Node","B1","B2","B3","B4","B5","B6","Flower")

#"#7bc437"
green.red <- colorRampPalette(c("#5F6D54","#CAB393","#BF7F58"))(8)
#"#c24a3a","#522c28"
i=10
system("mkdir barplot_all")
for (gene in rownames(per_condition_means)){
print(gene)
limitOfY <- max(per_condition_means[gene,]+per_condition_sd[gene,]/sqrt(6))*1.2
pdf(paste0("./barplot_all/barplot",gene,".pdf"),h=10,w=17)
barplot <- barplot(per_condition_means[gene,],beside=T,col=green.red, yaxt='n',xaxt='n',ylim=c(0,limitOfY),main =paste("Expression of",gene),cex.main=2.3,width=1.2,space=.9)

#space=c(.2,0,0,0,.2,0,0,0,.2,0,0,0,.2,0,0,0)

title(ylab="CPM",cex.lab=1.5,line=2.3)

#legend("topleft",legend=c("Node","B1", "B2","B3","B4","B5","B6", "Flower"),col = green.red,pch=19,cex=1.5)

if( limitOfY <= 50 ){
i=3
}
if( limitOfY > 50 & limitOfY <= 100 ){
i=10
}
if( limitOfY > 100 & limitOfY <= 1000 ){
i=50
}
if( limitOfY > 1000 & limitOfY <= 10000 ){
i=100
}
if( limitOfY > 10000 & limitOfY <= 100000 ){
i=1000
}
if( limitOfY > 100000 & limitOfY <= 1000000 ){
i=10000
}
if( limitOfY > 1000000 & limitOfY <= 10000000 ){
i=100000
}

axis(2,seq(0,limitOfY,i),labels=F)
mtext(side=2,text=seq(0,limitOfY,i),outer=F,las=2,line=.8,at=seq(0,limitOfY,i),cex=.8)


mtext(side=1,text=c("Node","B1","B2","B3","B4","B5","B6","Flower"),outer=F,line=1.2,at=barplot,cex=2)


arrows(x0 = barplot, y0 = per_condition_means[gene,] - per_condition_sd[gene,]/sqrt(5), x1 = barplot, y1=per_condition_means[gene,] + per_condition_sd[gene,]/sqrt(5) ,code=3,angle=90,length=0.05,col="darkgreen",lwd=4)

dev.off()

}


save.image("DEsmallRNAseq.RData")


miRsExpression<- rowSums(cpm.matrix[grep("mir|Mir|miR|novel",rownames(cpm.matrix)),])

ordered.miRsExpression <- miRsExpression[order(miRsExpression,decreasing=F)]

top60ordered.miRsExpression <- head(ordered.miRsExpression,60)

top60ordered.miRsExpression <- top60ordered.miRsExpression[order(top60ordered.miRsExpression,decreasing=F)]

pdf("miRexpression.pdf",w=6,h=20)
par(mar=c(4,10,4,4))

barplot <- barplot(log(ordered.miRsExpression),xlab="",ylab="",main="log of CPM",horiz=T,col = "grey40", space=1.8,yaxt='n',xaxt='n',xlim=c(0,20))

axis(side = 2, at=barplot, cex.axis = 0.7, labels=names(ordered.miRsExpression),las=2)

axis(side = 3, at=c(0,5,10,15,20), cex.axis = 1, labels=c(0,5,10,15,20),las=1,line=-3)

axis(side = 1, at=c(0,5,10,15,20), cex.axis = 1, labels=c(0,5,10,15,20),las=1,line=-3)

#abline(v=c(5,10,15,20))

dev.off()

system("open miRexpression.pdf")

#########################

pdf("miRexpressionNoNORM.pdf",w=6,h=20)
par(mar=c(4,10,4,4))

barplot <- barplot(ordered.miRsExpression,xlab="",ylab="",main="CPM",horiz=T,col = "grey40", space=1.8,yaxt='n',xaxt='n',xlim=c(0,25000000))

axis(side = 2, at=barplot, cex.axis = 0.7, labels=names(ordered.miRsExpression),las=2)

axis(side = 3, at=c(0,5000000,10000000,15000000,25000000), cex.axis = 1, labels=c("0","5M","10M","15M","25M"),las=1,line=-3)

axis(side = 1, at=c(0,5000000,10000000,15000000,25000000), cex.axis = 1, labels=c("0","5M","10M","15M","25M"),las=1,line=-3)

#abline(v=c(5,10,15,20))

dev.off()

system("open miRexpressionNoNORM.pdf")

save.image("DEsmallRNAseq.RData")

#################################
Mir156/172 expression

cpm.matrix.noNorm <- cpm(analysis_matrix,normalized.lib.sizes=F)

cpm.matrix.noNorm<- cpm.matrix.noNorm[,reorderColuns]

miR156Family <- colSums(cpm.matrix[grep("miR156",rownames(cpm.matrix)),])
miR172Family <- colSums(cpm.matrix[grep("miR172",rownames(cpm.matrix)),])

bp <- barplot(miR156Family,col ="#976464", space=1,xaxt='n')

mtext(side =1,at=bp,text= colnames(cpm.matrix),las=2, line=.5,cex=0.7)

bp <- barplot(miR172Family,col ="#698595", space=1,xaxt='n')

#plot(log(miR156Family) ~ log(miR172Family))
#################################
###########Blast for TAS #########
###################################


blastn -task blastn-short -query miR390.fasta -db ../PHAS > Target.miR390.blast.txt

blastn -task blastn-short -query cc.tsA.mir390.fasta -db ../PHAS > cc.tsA.mir390.blast.txt
blastn -task blastn-short -query cc.tsB.mir390.fasta -db ../PHAS > cc.tsB.mir390.blast.txt
blastn -task blastn-short -query cc.tsC.mir390.fasta -db ../PHAS > cc.tsC.mir390.blast.txt
blastn -task blastn-short -query cc.tsD.mir390.fasta -db ../PHAS > cc.tsD.mir390.blast.txt
blastn -task blastn-short -query cc.tsE.mir390.fasta -db ../PHAS > cc.tsE.mir390.blast.txt
blastn -task blastn-short -query cc.tsF.mir390.fasta -db ../PHAS > cc.tsF.mir390.blast.txt
blastn -task blastn-short -query cc.tsG.mir390.fasta -db ../PHAS > cc.tsG.mir390.blast.txt
blastn -task blastn-short -query cc.tas3ARFa.fasta -db ../PHAS > cc.tas3ARFa.blast.txt
blastn -task blastn-short -query cc.tas3ARFb.fasta -db ../PHAS > cc.tas3ARFb.blast.txt
blastn -task blastn-short -query cc.tas3ARFc.fasta -db ../PHAS > cc.tas3ARFc.blast.txt

###########################

#>TAS4si
#ACTTCCTAGCTCCAGCTCCGT
#>ceu-miR828b-5p
#TCTTGCTCAAATGAGTATTCCA
#TS.miR828
#rv.miR828
#TGGAATACTCATTTGAGCAAGA

blastn -task blastn-short -query miR828.fasta -db ../PHAS > miR828.blast.txt
blastn -task blastn-short -query TS.miR828.fasta -db ../PHAS > TS.miR828.blast.txt
blastn -task blastn-short -query RC.miR828.fasta -db ../PHAS > RC.miR828.blast.txt

############################

blastn -task blastn-short -query TAS1.2.si.fasta -db ../PHAS > TAS1.2.si.fasta.blast.txt

###########################

blastn -task blastn-short -query mir6026andMIR10533.fasta -db ../PHAS > mir6026andMIR10533.blast.txt

blastn -task blastn-short -query mir6026andMIR10533.fasta -db ../CArabicaPrecursorMir > mir6026andMIR10533.X.ALLMIRblast.txt
